### DISCUSSION TITLE
🆚打造舒服的vscode~幾個實用的插件&個人化設定

### DISCUSSION THREAD

**Discussion by JunTingLin** [1101-GS4719-C/Python-Community-GS4719-C#520] [General]
# 🆚打造舒服的vscode~幾個實用的插件&個人化設定

## vscode是啥?是Visual Studio嗎❓
我們一般講的vscode是指Visual Studio code，他的icon是藍色的，跟整合開發環境IDE的Visual Studio，icon是紫色的不同

![GitHub](https://i.ytimg.com/vi/d8FsNESkE0M/maxresdefault.jpg)

Visual Studio Code是一款由微軟開發且跨平台的免費原始碼編輯器。該軟體支援語法突顯、程式碼自動補全、程式碼重構功能，並且內建了命令列工具和 Git 版本控制系統，可以直接用圖形化的方式，pull、add、commit、push，重點是介面長得好看又輕量化，不像某些IDE又大又肥🐷(但對啦~畢竟vscode只能算是強大的原始碼編輯器)

## 整條龍一次包辦的IDE也是有他的好，但...😅
寫python常用到的PyCharm、寫C++用到的Visual Studio、寫java常用到的Eclipse、NetBeans，都是一條龍包辦的整合開發環境，包含編輯器、Debuger、圖形介面設計工具、打包發行工具，整條龍都幫你用好，不須轉來轉去，但...就如上面提的，他們肥~~可以客製化的介面功能又少

## 看吧~最受歡迎的是它💥
根據[stackoverflow 2021的數據統計](https://insights.stackoverflow.com/survey/2021#most-popular-technologies-new-collab-tools)，八萬個受訪者中有超過7成都選用vscode，當作開發環境

<img width="1440" alt="截圖 2022-03-26 下午5 07 43" src="https://user-images.githubusercontent.com/92431095/160232741-4ceeb730-6140-4935-8418-208cb3dcd9df.png">

圖中它把vscode當作IDE，嗯....嚴格說也是啦，當你越裝越多插件，就幾乎越來越可以跟IDE媲美了

## 最推薦通用差件💪

 1. [vscode-icons](https://marketplace.visualstudio.com/items?itemName=vscode-icons-team.vscode-icons)
 漂漂亮亮的icon
![GitHub](https://raw.githubusercontent.com/vscode-icons/vscode-icons/master/images/screenshot.gif)

 2. [Git Graph](https://marketplace.visualstudio.com/items?itemName=mhutchie.git-graph)
 版本控制直接圖形show給你
 ![GitHub](https://github.com/mhutchie/vscode-git-graph/raw/master/resources/demo.gif)

 3. [Chinese (Traditional) Language Pack for Visual Studio Code](https://marketplace.visualstudio.com/items?itemName=MS-CEINTL.vscode-language-pack-zh-hant)
 有人喜歡英文那沒差🤣~

 4. [indent-rainbow](https://marketplace.visualstudio.com/items?itemName=oderwat.indent-rainbow)以色塊顯示縮排
   ![GitHub](https://pbs.twimg.com/media/D_lF-abUYAAiBLl.jpg:large)

 5. [Bookmarks](https://marketplace.visualstudio.com/items?itemName=alefragnani.Bookmarks)如書籤功能一樣，將任一行加入書籤，列表於左側，可透過書籤快速跳至行
   ![GitHub](https://github.com/alefragnani/vscode-bookmarks/raw/HEAD/images/printscreen-toggle.png)

## 結語
上面列的是比較通用的，插件當然還有很多比如寫網頁會需要html auto rename tag，查詢色票會需要Vscode Pigments，快速搜尋CDN的cdnjs，反正很多啦!差件真的超好用的，大家都有啥插件呢?可以在下方回覆喔😁

**Comment by JunTingLin** [comment]
😊順帶一提vscode還可以去個人化的設定與配置，有分
+ 全域的使用者設定:會應用於該使用者開啟的所有工程
+ 工作空間設定:工作空間是指使用VS Code開啟的某個資料夾，在該資料夾下會建立一個名為.vscode的隱藏資料夾，裡面包含著僅適用於當前目錄的VS Code的設定

**Reply by chia-kai-chang** [reply]
不同的專案可以使用不同的vscode喜好設定，很方便

**Comment by 11timyyy** [comment]
把介面弄得漂亮程式寫起來也會感到很舒服~

**Comment by HsuChen88** [comment]
謝謝你每次都提供很有幫助的資訊

**Reply by chia-kai-chang** [reply]
@HsuChen88 要不要嘗試提供一些程式開發新知😄

**Comment by henrywen547** [comment]
之前可能就只是利用工具列調整一些設定跟外觀，
謝謝你讓我了解到讓vscode變得更美觀的方法!

**Comment by ThomasMeng0515** [comment]
vscode除了開發環境很舒服以外，裡面有許多快捷鍵也滿方便的，像是 shift + alt + F 可以自動排版程式碼，不用自己排版~
快捷鍵的說明可以參考:
https://blog.poychang.net/vscode-shortcuts/

**Comment by JSHT** [comment]
常會用到的顏色插件
[Color Highlight](https://marketplace.visualstudio.com/items?itemName=naumovs.color-highlight)
![image](https://user-images.githubusercontent.com/48956859/160344817-9a6783c7-c76e-4689-a388-4e4eeaea0a23.png)
如圖，當打上色碼，就會幫你加上對應的底色。

**Reply by JSHT** [reply]
另外也推薦可以進行配對括號的顏色設定
[vscode 內建 bracket colorizer](https://blog.kyomind.tw/bracket-pair-colorization/)

**Comment by hapiwangy** [comment]
感謝分享這些插件讓之後我在vs code的使用上更佳的方便。

**Comment by leon10456** [comment]
我還推薦ESlint 能自動排版及修正錯字的插件 很好用

**Comment by Ku-Hao-Chen** [comment]
謝謝分享，版面變得很舒服。

**Comment by PoooYen** [comment]
能在生硬的程式語言裡面增加美感真的是很舒服的事情

**Comment by Yuhsin000** [comment]
之前一直在想說藍色的跟紫色的差在哪裡，原來是這樣，之後來美化他~~

**Comment by sean1325418** [comment]
現在才知道有這麼多可以安裝的東西，弄了之後看起來舒服多了

**Comment by d26488399** [comment]
謝謝分享，我一支知道有很多空衝功能可以安裝，但沒有方向，看完這篇比較有方向了

**Comment by treeann** [comment]
覺得vscode真的很方便 
也都是用vscode來打python 
但是最近想拿來打c++一直出問題，無法編譯難受香菇

**Comment by SamTU0222** [comment]
謝謝分享，vscode擴充功能真的是程式新手的救星

**Comment by ghost** [comment]
謝謝分享，雖然不一定會用到，但這樣看起來是真的舒服

**Comment by JunTingLin** [comment]
這邊在補充並分享一個AI判斷你下一步要coding啥的插件⏬
[😲這太強啦!AI幫你寫程式(最牛插件).md #544](https://github.com/NCU-GS4719-Python/Python-Community/discussions/544)

**Comment by HaoFish123** [comment]
感謝分享

**Comment by yi-1111** [comment]
這樣打code起來很舒服耶

**Comment by etin2468** [comment]
五彩繽紛的介面感覺很讚

**Comment by Rosancl** [comment]
感謝分享~~用了icon之後變好看很多 👍 👍

**Comment by jason79461385** [comment]
用vim用久了看到五顏六色的反而會有點不習慣 哈哈哈哈哈 但還是感謝分享

**Comment by liangyinglly** [comment]
套用後版面好繽紛
